docs/fassets/developer-guides/12-fassets-autoredeem.mdx

```diff
@@ -0,0 +1,649 @@
+---
+title: FAsset Auto-Redemption
+tags: [intermediate, fassets]
+slug: fassets-autoredeem
+description: Learn how to bridge FAssets between Flare and Hyperliquid EVM Testnet
+keywords: [fassets, flare-network]
+sidebar_position: 12
+---
+
+import CodeBlock from "@theme/CodeBlock";
+import FassetRedeemComposer from "!!raw-loader!/examples/developer-hub-solidity/FAssetRedeemComposer.sol";
+import bridgeToHyperEVM from "!!raw-loader!/examples/developer-hub-javascript/bridgeToHyperEVM.ts";
+import autoRedeemHyperEVM from "!!raw-loader!/examples/developer-hub-javascript/autoRedeemHyperEVM.ts";
+import getOftPeers from "!!raw-loader!/examples/developer-hub-javascript/getOftPeers.ts";
+
+# FAsset Auto-Redemption
+
+## Introduction
+
+This documentation covers a cross-chain bridge system that allows you to move FAssets (tokenized versions of non-smart contract tokens like XRP) between Flare Network's Coston2 testnet and Hyperliquid EVM Testnet.
+The system goes beyond simple token bridging by enabling **automatic redemption** - you can send wrapped XRP (FXRP) from Hyperliquid and have it automatically converted back to native XRP on the XRP Ledger, all in a single transaction.
+
+**What you can do:**
+
+- Bridge FXRP tokens from Coston2 to Hyperliquid EVM to prepare for auto-redemption
+- Send FXRP from Hyperliquid back to Coston2 and auto-redeem to native XRP
+- Execute the entire bridge-and-redeem flow with just one transaction
+
+**Key technologies:**
+
+- LayerZero OFT (Omnichain Fungible Token) for cross-chain messaging
+- Flare's FAsset system for tokenizing non-smart contract assets
+- Composer pattern for automatic execution on the destination chain
```

Comment:
```suggestion
- LayerZero OFT (Omnichain Fungible Token) for cross-chain messaging.
- Flare's [FAsset](/fassets/overview) system for tokenizing non-smart contract assets.
- Composer pattern for automatic execution on the destination chain.
```

* Add link to LayerZero OFT to LayerZero docs.
* Is it LayerZero composer, needs a bit more explanation what is it.